### PR TITLE
Remove use of cache when cache_active is False

### LINKED ISSUE #7332: Argument 'cache' of skimage.measure.regionprops_table (and regionprops) doesn't seem to have any effect
### Description:

When passing cache=False to regionprops_table (or regionprops directly), it seems that properties still get cached in RegionProperties._cache. 
I think that an easy solution would be to return f(obj) when obj._cache_active is False in '_cached' decorator.

### Way to reproduce:

```python
import numpy as np
import skimage as ski

a = np.array([[0,0], [0,1]])

with_cache = ski.measure.regionprops(a, cache=True)[0]
with_cache.perimeter # acces one property to get data into cache

without_cache = ski.measure.regionprops(a, cache=False)[0]
without_cache.perimeter # acces one property to get data into cache

assert with_cache._cache != {} # expected
assert without_cache._cache == {} # will fail

```

### Version information:

_No response_

### PR DESCRIPTION
**Pull request by Maxime-corbe**
Cache dictionary was used even if cache_active was set to False. It was computed every time and still use extra memory to store the result. I propose here to remove completely the use of this attribute when cache_active is False.
    
Resolve #7332

## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

## Release note

Summarize the introduced changes in the code block below in one or a few sentences. The
summary will be included in the next release notes automatically:

```release-note
Ensure cache stays empty when `cache=False` is passed to `skimage.measure.regionprops`.
```

### CODE REVIEW

**Comment by mkcor** [comment]
Thank you, @Maxime-corbe! Do you think you could add a small test for this fix?

File: skimage/measure/_regionprops.py [hunk 1]
```diff
@@ -235,7 +235,10 @@ def wrapper(obj):
...
         prop = f.__name__
 
-        if not ((prop in cache) and obj._cache_active):
+        if not obj._cache_active:
+            return f(obj)
+
+        if prop not in cache:
             cache[prop] = f(obj)
 
         return cache[prop]
```
Comment (lagru) [inline skimage/measure/_regionprops.py:244]: How about
```suggestion
        out = f(obj)
        if obj._cache_active and prop not in cache:
            cache[prop] = f(obj)
        return out
```
That logic might be even easier to follow?

File: skimage/measure/tests/test_regionprops.py [hunk 2]
```diff
@@ -1236,6 +1236,16 @@ def test_cache():
     assert np.any(f0 != f1)
 
 
+def test_cache_memory():
+    SAMPLE_mod = SAMPLE.copy()
+    region = regionprops(SAMPLE_mod, cache=False)[0]
+    # Acces one property
```
Comment (lagru) [inline skimage/measure/tests/test_regionprops.py:1242]:
```suggestion
    # Access one property to trigger cache
```

File: skimage/measure/tests/test_regionprops.py [hunk 3]
```diff
@@ -1236,6 +1236,16 @@ def test_cache():
...
+    _ = region.image_filled
+
+    # Cache should be empty
+    assert not region._cache
```
Comment (lagru) [inline skimage/measure/tests/test_regionprops.py:1246]: A bit more expressive:

```suggestion
    assert region._cache == dict()
```

File: skimage/measure/tests/test_regionprops.py [hunk 4]
```diff
@@ -1236,6 +1236,16 @@ def test_cache():
     assert np.any(f0 != f1)
 
 
+def test_cache_memory():
```
Comment (lagru) [inline skimage/measure/tests/test_regionprops.py:1239]:
```suggestion
def test_disabled_cache_is_empty():
```

**Review by lagru — APPROVED** [review]
Thanks @Maxime-corbe! A few nitpicks but looks good otherwise.

File: skimage/measure/_regionprops.py (hunk 1 above)
Comment (Maxime-corbe) [reply skimage/measure/_regionprops.py:244]: If you do that, f(obj) is computed every time, even if it's already in the cache (and cache is active).

Comment (lagru) [reply skimage/measure/_regionprops.py:244]: Oh, I entirely missed that context! Never mind then and thanks for pushing back.

**Comment by lagru** [comment]
Good to go in my opinion. Thanks @Maxime-corbe!

**Review by mkcor — APPROVED** [review]
Thank you for adding a test, @Maxime-corbe! :pray:

**Comment by mkcor** [comment]
@Maxime-corbe fyi, you may commit multiple suggestions at once, i.e., as a batch.

Reference: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes

**Pull request merged**